widgetry/src/backend_glow_native.rs

```diff
@@ -12,51 +12,60 @@ pub fn setup(window_title: &str) -> (PrerenderInnards, winit::event_loop::EventL
     // TODO If people are hitting problems with context not matching what their GPU provides, dig up
     // backend_glium.rs from git and bring the fallback behavior here. (Ideally, there'd be
     // something in glutin to directly express this.) multisampling: 2 looks bad, 4 looks fine
-    let context = match glutin::ContextBuilder::new()
+    let context = glutin::ContextBuilder::new()
         .with_multisampling(4)
         .with_depth_buffer(2)
-        .build_windowed(window, &event_loop)
-    {
-        Ok(ctx) => ctx,
-        Err(err) => {
-            panic!("Your videocard doesn't support the OpenGL mode requested. This is a common issue when running inside a virtual machine; please run natively if possible. See https://github.com/dabreegster/abstreet/issues/103 for more info, and feel free to ask for help using that issue.\n\nError: {}", err);
-        }
-    };
+        .build_windowed(window.clone(), &event_loop).map_err(|err| {
+            warn!("Trying default graphics context after standard graphics context failed with error: {:?}",  err);
+            glutin::ContextBuilder::new().build_windowed(window.clone(), &event_loop)
+    })
+        .map_err(|err| {
+            warn!("Trying graphics context with vsync after default graphics context failed with error: {:?}", err);
+            glutin::ContextBuilder::new()
```

Comment:
> Did not work - panicked at runtime. See gist.

It seems like this code here is returning an Err - but to my eye it looks like the same code from your fork which passed, weird. 👀

Reply:
Wait, [map_err](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err) isn't the right function here. This is actually returning `Result<Context, Result<Context, Error>>`! I think we want `or_else` in both spots.

If this fix works, I'd like to merge it -- this is a way simpler approach than diving into `glutin` and I think it'd help other people.

Reply:
oops, that's a common mistake for me. `or_else` makes sense, and I think explains the error @gedkott was seeing.

I've fixed, rebased, and re-pushed.

@gedkott - any chance you could try building the latest version of this branch from your setup, and forward along any logging if things don't work.

Reply:
Yessir. I will be unavailable until later this weekend or next week. I'll check then.